### PR TITLE
refactor: Return CAutoFile from BlockManager::Open*File()

### PR DESCRIPTION
**Pull request by maflcko**
This is required for https://github.com/bitcoin/bitcoin/pull/28052, but makes sense on its own, because offloading logic to `CAutoFile` instead of re-implementing it allows to delete code and complexity.

### CODE REVIEW

File: src/streams.h [hunk 1]
```diff
@@ -598,9 +596,9 @@ class BufferedFile
             readNow = nAvail;
         if (readNow == 0)
             return false;
-        size_t nBytes = fread((void*)&vchBuf[pos], 1, readNow, src);
+        size_t nBytes{m_src.detail_fread(Span{vchBuf}.subspan(pos,readNow))};
```
Comment (TheCharlatan) [inline src/streams.h:599]: Nit (clang-format): Space after `pos,`.

File: src/bench/streams_findbyte.cpp [hunk 2]
```diff
@@ -10,21 +10,21 @@
...
-    FILE* file = fsbridge::fopen("streams_tmp", "w+b");
+    CAutoFile file{fsbridge::fopen("streams_tmp", "w+b"), 0};
     const size_t file_size = 200;
     uint8_t data[file_size] = {0};
     data[file_size-1] = 1;
-    fwrite(&data, sizeof(uint8_t), file_size, file);
-    rewind(file);
-    BufferedFile bf{file, /*nBufSize=*/file_size + 1, /*nRewindIn=*/file_size, 0};
+    file << data;
+    std::rewind(file.Get());
```
Comment (TheCharlatan) [inline src/bench/streams_findbyte.cpp:18]: Nit: Include `cstdio` where this is used?

**Review by sedited — APPROVED** [review]
ACK fa33b6fba270e5867f208b242f50145e1ed3550e

**Review by sedited — APPROVED** [review]
Re-ACK fa56c421be04af846f479c30749b17e6663ab418

Just addressing nits.

File: src/validation.cpp [hunk 3]
```diff
@@ -4589,8 +4589,7 @@ void ChainstateManager::LoadExternalBlockFile(
 
     int nLoaded = 0;
     try {
-        // This takes over fileIn and calls fclose() on it in the BufferedFile destructor
```
Comment (kevkevinpal) [inline src/validation.cpp:4592]: did you mean to remove this comment?

**Review by willcl-ark — APPROVED** [review]
tACK fa56c421be

Clean refactor needed for future work.

**Pull request merged**